src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -1646,11 +1647,11 @@ class CaseExpressionSegment(BaseSegment):
                 OneOf(
                     Ref("SignedSegmentGrammar"),
                     # Ref('TildeSegment'),
-                    "NOT",
+                    Ref("NotKeywordSegment"),
                     "PRIOR",
                     # used in CONNECT BY clauses (EXASOL, Snowflake, Postgres...)
                 ),
-                Ref("Expression_C_Grammar"),
+                Ref("Expression_A_Grammar"),
```

Comment:
Let's keep this out of this PR.

Reply:
Indeed. I was initially thinking that this was important for the MySQL fix. But it's not.
Whether it's a good idea or bad one, it doesn't belong in this PR.